### DISCUSSION TITLE
federation support?

### DISCUSSION THREAD

**Discussion by omar-dulaimi** [MichalLytek/typegraphql-prisma#203] [Q&A]
Hello,

Please excuse me if this question has already been answered somewhere. 

I want to know if the generated resolvers and types are federation ready or not. And if it's impossible to implement federation unless it's done by hand manually for all models.

Because I saw the use of directives and other stuff on the custom resolvers in the example.

**Comment by MichalLytek** [comment]
No, you need to use the `enhance` function helpers in order to add required federation metadata/directives.

Prisma does not know about your other federated services.

**Reply by omar-dulaimi** [reply]
Is it possible to get the federation example integrated with typegraphql-prisma?
It's a little bit confusing with all these resolvers and meta data.
Is this the `enhance function helpers` you meant?
https://prisma.typegraphql.com/docs/advanced/additional-decorators

Because it does not mention anything about metadata and directives; just decorators.

I thought ModelsEnhanceMap is what I needed, but nothing was shown in the generated files.

**Reply by MichalLytek** [reply]
Enhance works in runtime, not in generation time, so the generators are applied in memory, not in files.

So you need to apply the same decorator that you would place manually on resolvers or types, like in examples:
https://github.com/MichalLytek/type-graphql/tree/master/examples/apollo-federation

**Reply by omar-dulaimi** [reply]
@MichalLytek Just to check, the metadata are added using extensions right?

**Reply by MichalLytek** [reply]
@omar-dulaimi Have you read the examples?
```
@Directive(`@key(fields: "id")`)
```

**Reply by omar-dulaimi** [reply]
@MichalLytek I did, but where did you get that snippet from though?

**Reply by omar-dulaimi** [reply]
I mean, in the context of enhance.

**Reply by omar-dulaimi** [reply]
```
  const modelsEnhanceMap: ModelsEnhanceMap = {
    Product: {
      class: [
        Directive(`@key(fields: "id")`),
      ],
    },
  };

```
